### DISCUSSION TITLE
Is it expected to affect the result by assigning different value to solver.parameters.random_seed?

### DISCUSSION THREAD

**Discussion by maverick-xinwei** [google/or-tools#2474] [CP-SAT questions] [answered]
I did the experiment below but got the same results from running the script multiple times.  Is there something else I should modify to make the solver produce different results  when different seeds are given ? 

Another thing is , I always get OPTIMAL as the solver's status at the end of the code . I thought the model I specified here is not really a optimization problem, i.e. there's no definition of which solution is more optimized than another.  So, what does OPTIMAL really mean in this case ?  Or, is it so that it always return OPTIMAL when running a non-optimization problem because it thought every solution is equally optimized ? 

Many thanks !

```
1 
  2 from ortools.sat.python import cp_model
  3 import random
  4 
  5 model = cp_model.CpModel()
  6 
  7 x = model.NewIntVar(-256,255, 'x')
  8 y = model.NewIntVar(-256,255, 'y')
  9 z = model.NewIntVar(0,255, 'z')
 10 
 11 model.Add(x+y+z==100)
 12 model.Add(2*x+3*y+4*z<100)
 13 #model.AddHint(x, 1)
 14 #model.AddHint(y, 2)
 15 #model.AddHint(z, 3)
 16 
 17 solver = cp_model.CpSolver()
 18 solver.parameters.random_seed = random.randint(0, 2**16)
 19 solver.parameters.cp_model_presolve =  False
 20 
 21 print("seed = ", solver.parameters.random_seed)
 22 
 23 
 24 #solution_printer = cp_model.VarArrayAndObjectiveSolutionPrinter([x,y,z])
 25 #status = solver.SolveWithSolutionCallback(model, solution_printer)
 26 #status = solver.SearchForAllSolutions(model, solution_printer)
 27 
 28 status = solver.Solve(model)
 29 
 30 print ("status = %s"% solver.StatusName(status))
 31 print ("objective = %i"% solver.ObjectiveValue())
 32 print ("x = ", solver.Value(x))
 33 print ("y = ", solver.Value(y))
 34 print ("z = ", solver.Value(z))
 35 #print (model.Proto())       
```

**Comment by lperron** [comment]
In case of models without objective, OPTIMAL means a solution was found.
This is the convention from the MIP community.

The random seed has some impact on the search. But in your case, it is likely that the hint overrides any randomization.

**Reply by maverick-xinwei** [reply]
Thanks for the reply.  

The hints were actually commented out.

**Comment by lperron** [comment]
You can try hinting different random values.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le lun. 5 avr. 2021 à 21:55, maverick_xinwei ***@***.***> a
écrit :

**Reply by maverick-xinwei** [reply]
Is it guaranteed that cp-sat will always produce the same results with same hint values ?

In what case does parameters.random_seed affect the randomization ?

**Comment by lperron** [comment, marked as answer]
1) in sequential mode, yes with the same seed.
2) it affects tie breaking of variables. It does not have a huge effect on
the branching schema of the search tree.

Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le mar. 6 avr. 2021 à 11:02, maverick_xinwei ***@***.***> a
écrit :

**Reply by maverick-xinwei** [reply]
Thanks !
